src/app/pages/add-network/add-network.tsx

```diff
@@ -55,17 +53,13 @@ export const AddNetwork = () => {
             const chainInfo = await response.json();
             const networkId = chainInfo?.network_id && parseInt(chainInfo?.network_id);
             if (networkId === ChainID.Mainnet || networkId === ChainID.Testnet) {
-              setNetworks(networks => {
-                return {
-                  ...networks,
-                  [key]: {
-                    url: origin,
-                    name,
-                    chainId: networkId,
-                  },
-                };
+              networksActions.newNetworkAdded({
+                chainId: networkId,
+                id: key,
+                name,
+                url: origin,
               });
-              setNetworkKey(key);
+              networksActions.currentNetworkChanged(key);
```

Comment:
Do we ever allow users to add a network and _not_ change to it by default? If not, change the network inside `newNetworkAdded`